recipes/cgal/all/conanfile.py

```diff
@@ -86,23 +64,13 @@ def package(self):
         self.copy("LICENSE*", dst="licenses", src=self._source_subfolder)
         cmake = self._configure_cmake()
         cmake.install()
-        tools.rmdir(os.path.join(self.package_folder, "share"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-        if self.options.get_safe("shared"):
-            for root, _, filenames in os.walk(os.path.join(self.package_folder, "bin")):
-                for filename in filenames:
-                    if not filename.endswith(".dll"):
-                        os.unlink(os.path.join(root, filename))
-        else:
-            tools.rmdir(os.path.join(self.package_folder, "bin"))
+        files.rmdir(self, os.path.join(self.package_folder, "share"))
+        files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
+        files.rmdir(self, os.path.join(self.package_folder, "bin"))
 
     def package_info(self):
-        # TODO: add components
         self.cpp_info.names["cmake_find_package"] = "CGAL"
         self.cpp_info.names["cmake_find_package_multi"] = "CGAL"
```

Comment:
Hi, I do not fully understand your suggestion.

- Actually, I understand the suggestion about `self.cpp_info.set_property`: that refers to https://docs.conan.io/en/latest/migrating_to_2.0/properties.html
- I do not understand your reference to https://github.com/CGAL/cgal/blob/d49ee20965ad41ba9e476c5e92fa36a3d45b3be4/Installation/CMakeLists.txt#L774-L807.

Reply:
> I do not understand your reference to ...

99% was just for my future self since I look it up

***

The CMake installation does not include any `TARGETS` or `EXPORTS` of a target group, just lots of scripts so I was not sure where the last name came from

But now I noticed https://github.com/CGAL/cgal/blob/d49ee20965ad41ba9e476c5e92fa36a3d45b3be4/Installation/lib/cmake/CGAL/CGALConfig.cmake#L169 and https://github.com/CGAL/cgal/blob/d49ee20965ad41ba9e476c5e92fa36a3d45b3be4/Installation/lib/cmake/CGAL/CGALConfig.cmake#L8

Reply:
OK, now I get the idea. The CGAL CMake system does not use `export(TARGETS)` and/or `install(EXPORT...)`. The imported targets (`CGAL::CGAL`, `CGAL::CGAL_Core`, `CGAL::CGAL_ImageIO`, and `CGAL::CGAL_Qt5`) are created directly in the file `CGALConfig.cmake`, that I wrote myself.

CGAL CMake system has a long history, that started with CMake 2.2 in 2007, and at that time CGAL was not header only, and there was 4 compiled libraries in addition to the set of C++ templates.

I keep trying to simplify all that, from time to time. Now that CGAL requires a fairly new version of CMake (>= 3.14), I think I could switch to a simpler logic, more mainstream.


So, your suggestion was only about the use of `self.cpp_info.set_property`, and the quote of `CMakeLists.txt` was only about the doubt about the name of the imported target.

Reply:
Thanks for filling me in, that makes a lot knowing the history, I only started programming in 2011 and CMake in ~2016 so anything older makes no sense 🙃 

Amazing work keeping this project maintained btw, everyone using this certainly appreciates you ❤️